lib/src/kbasesearchengine/main/NarrativeInfoDecorator.java

```diff
@@ -41,9 +43,12 @@
  */
 public class NarrativeInfoDecorator implements SearchInterface {
 
+    private final Logger logger = LoggerFactory.getLogger(NarrativeInfoDecorator.class);
```

Comment:
Don't make a logger instance. There's some weird effects that I haven't totally figured out, or understand, regarding making a reference to a logger and OOM errors.

Instead do:

```
private Logger getLogger() {
    return LoggerFactory.getLogger(...);
}
```
and use that.

Reply:
Isn't LoggerFactory.getLogger(...) making an instance?

Reply:
Yes sorry, I should've said don't memoize a logger instance in this class, which creates a GC link to the logger instance.